src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -177,6 +179,7 @@ protected override void Apply(TransactionOperationContext context, BlittableJson
 
                         SetValueForTypedDatabaseCommand(context, type, cmd, index, leader, out object result);
                         leader?.SetStateOf(index, result);
+                        IndexChangedForBackup?.Invoke(this, (cmd, index, type));
```

Comment:
This should be only invoked when transaction is committed (e.g. check SetValueForTypedDatabaseCommand) and I think it would be better to invoke this for all commands and filter it out there? You need only databaseName, index and type, not BlittableJSON